### PR TITLE
Fix has_many forms index

### PR DESCRIPTION
**Pull request by dhyegocalota**
(see https://github.com/activeadmin/activeadmin/issues/5207)

### CODE REVIEW

**Comment by varyonic** [comment]
@caifara does this work for you? A test would be nice but I don't see any pre-existing feature or jasmine tests for has_many.es6 so I can't expect it.

File: app/assets/javascripts/active_admin/lib/has_many.es6 [hunk 1]
```diff
@@ -41,7 +41,18 @@ $(function() {
...
     if (!before_add.isDefaultPrevented()) {
-      let index = parent.data('has_many_index') || (parent.children('fieldset').length - 1);
+      let index = parent.data('has_many_index');
+
+      if (!index) {
+        var lastInputIndex = parent.children('fieldset')
+          .last()
+          .find('input')
+          .prop('name')
+          .match(/^.*(?:\[.*\])*\[([0-9])+\]\[.*\]$/);
```
Comment (caifara) [inline app/assets/javascripts/active_admin/lib/has_many.es6]: moved the +-sign to allow for an index >= 10
````js
.match(/^.*(?:\[.*\])*\[([0-9]+)\]\[.*\]$/);

**Comment by dhyegofernando** [comment]
@caifara I've refactored like the way you suggested. Thanks 👍

**Comment by varyonic** [comment]
Please rebase and reconcile with #5196.

**Comment by deivid-rodriguez** [comment]
ping @dhyegofernando, can you get this working again?

**Comment by dhyegofernando** [comment]
Sure @deivid-rodriguez 
I'll do it

**Comment by javierjulio** [comment]
This has merge conflicts but if it can be updated we can accept. Thank you.

**Pull request closed without merging**